### ISSUE TITLE
schemaPollHeaders non functional

### ISSUE THREAD

**Issue by sjcobb2022** [HoudiniGraphql/houdini#549]
### Describe the bug

Hi there,

I've been spending time trying to load my own headers for an integration with shopify. To access the GraphQL database, you need an API key to be passed as a header. The schemaPollHeaders value is the only way to pass such headers, however it is currently non-functional.

https://github.com/HoudiniGraphql/houdini/blob/e02e524618e518fdf6fcbbb2596a86c9ab9dfe31/src/cmd/pullSchema.ts#L36-L40

The issue of the problem is that the schemaPollHeaders are not combined with the argument headers in the pullSchema function. 

The fix for this could be to combine the objects on line 39 of [src/cmd/pullSchema.ts](https://github.com/HoudiniGraphql/houdini/blob/main/src/cmd/pullSchema.ts#L39)

```js
// Write the schema
await pullSchema(
	config.apiUrl,
	config.schemaPath ? config.schemaPath : path.resolve(targetPath, 'schema.json'),
	{...headers, ...config.pullHeaders}
)
```

Bear in mind, running `houdini generate -h X-Header=TOKEN` works just fine.

### Severity

blocking all usage of Houdini

### Steps to Reproduce the Bug

1. Clone base project with recommended settings (e.g. `npx degit houdinigraphql/intro hello-houdini`)
2. Pass header through schemaPollHeaders in houdini.config.js. example below 
```js
const config = {
	client: './src/client',
	apiUrl: 'https://some.url.com/api/graphql',
	schemaPollHeaders: {
		'X-Necessary-Token-For-Auth': 'env:API_TOKEN'
	}
};
```
3. Unable to access graphql database. Given error: "⚠️  Couldn't pull your latest schema: Unexpected end of JSON input" 

Here is an example config that I believe still works.

```js
const config = {
	client: './src/client',
	apiUrl: 'https://hydrogen-preview.myshopify.com/api/2022-07/graphql.json',
	schemaPollHeaders: {
		'X-Shopify-Storefront-Access-Token': '3b580e70970c4528da70c98e097c2fa0'
	}
};
```

**Comment by AlecAivazis** [comment]
Thanks for submitting this @sjcobb2022! I'll get something out soon. I'm curious, is there a particular reason you are pulling the schema manually instead of letting the vite plugin do it for you?

**Comment by sjcobb2022** [comment]
> Thanks for submitting this @sjcobb2022! I'll get something out soon. I'm curious, is there a particular reason you are pulling the schema manually instead of letting the vite plugin do it for you?

I'm transitioning from kitQL so I thought i would gen manually to test things out in comparison to my old workflow. Without the apiUrl and headers in the houdini.config.js everything works fine.

I would also like to add that i believe that the schema is queried every 2000ms, which uses the config file. in my opinion the separation of the client and the config is a little obtuse and makes things harder for the developer, especially since one cannot use `import.meta.env` (or equivalent) in the houdini.config.json. without setting environment variables.

KitQL codegen combated this by using the ['--requires' directive](https://www.the-guild.dev/graphql/codegen/docs/config-reference/codegen-config#cli-flags) in their CLI. in this way, you could pass environment variables from .env files. `graphql-codegen --config ./.graphqlrc.yaml -r dotenv/config`.

TL;DR: Feature request for better parsing of the apiURL and headers would be much better. As having details in both the houdini.config.js and the client.ts is (imo) an over-abstraction.

**Comment by AlecAivazis** [comment]
Thanks for the feedback @sjcobb2022! I agree that this part of the API needs some refinement. 

Ideally, we would have a way to just import the file that users point us to but that doesn't work for typescript support package aliases so we're in kind of a rough place config-wise. We need some information for the plugin to function but we also need the fetch function to use to query to API. I wonder if we could just ask users to pass their fetch function to the config file? 🤔 

> Feature request for better parsing of the apiURL and headers would be much better. As having details in both the houdini.config.js and the client.ts is (imo) an over-abstraction. 

Sorry if I'm missing something obvious but what would this look like?

**Comment by sjcobb2022** [comment]
> > Feature request for better parsing of the apiURL and headers would be much better. As having details in both the houdini.config.js and the client.ts is (imo) an over-abstraction.
> 
> Sorry if I'm missing something obvious but what would this look like?

Generally I think the client.ts setup is fine. It works well and is very easily generated by `npx houdini init`

One "solution" that I was thinking of would be to do something a little different and have optional input to the pre-process function. Something along the lines of 

```typescript
import { sveltekit } from '@sveltejs/kit/vite';
import houdini from 'houdini/vite';
import { defineConfig, loadEnv } from 'vite';

const config = defineConfig(({ mode }) => {
	//loads env files e.g.
	const env = loadEnv(mode, process.cwd());

	return {
		plugins: [
			houdini({
				schemaPollHeaders: {
					'X-Shopify-Storefont-Acess-Token': env.VITE_SHOPIFY_STOREFRONT_ACCESS_TOKEN
				}
			}),
			sveltekit()
		]
	};
});
```
In theory, this could entirely make the use of the houdini.config.js useless, however both systems could be kept in place to provide more versatility. For example defining scalars is a lot easier in the houdini.config.js. However logically, any config provided in the pre-process function should be prioritize. E.g. 

```typescript
// vite.config.ts
houdini({
    headers: {
            X-Header: process.env.something
        }
})

// Versus

//houdini.config.js
const config = {
   header: {
        X-Header: 'string-or-something'
    }
}
```
In this scenario the vite config would be used.

During houdini init any headers specified in `npx houdini init -h` can be placed into both the client.ts and the vite.config.ts to ensure that any future queries have the necessary headers.
